modal/_utils/function_utils.py

```diff
@@ -459,6 +459,15 @@ def exc_with_hints(exc: BaseException):
 
 
 async def _process_result(result: api_pb2.GenericResult, data_format: int, stub, client=None):
+    try:
+        return await _process_result_wrap_exceptions(result, data_format, stub, client)
+    except _ContainerException as exc:
+        raise exc.unwrap()
+
+
+async def _process_result_wrap_exceptions(result: api_pb2.GenericResult, data_format: int, stub, client=None):
```

Comment:
Another option is to use a boolean flag to wrap exceptions, but that feels a bit more smelly to me